ctapipe/tools/tests/test_merge.py

```diff
@@ -107,13 +107,15 @@ def test_allowed_tels(tmp_path, dl1_file, dl1_proton_file):
 
     # create file to test 'allowed-tels' option
     output = tmp_path / "merged_allowed_tels.dl1.h5"
+
     ret = run_tool(
         MergeTool(),
         argv=[
             str(dl1_file),
             str(dl1_proton_file),
             f"--output={output}",
             "--allowed-tels=[1,2]",
+            "--split_datasets_by=tel_type",
```

Comment:
Why this change?

Reply:
One of those automatic code style anaylsis sent a warning (which worried me) that not all edit code in this PR was covered by the pytest, so I added the split also in the allowed tells test. Would it be better to have it separated in an additional test `def test_allowed_tels_and_split_datasets_by`?